src/coreComponents/dataRepository/Group.hpp

```diff
@@ -1283,6 +1284,13 @@ class Group
   integer getLogLevel() const { return m_logLevel; }
   ///@}
 
+  /**
+   * @brief Return PyGroup type.
+   * @return Return PyGroup type.
+   */
+  virtual PyTypeObject * getPythonType() const
```

Comment:
I don't really like how `Group.hpp` depends on stuff in `pygeosx`, but I suspect this is the type of stuff we will talk about in the meeting.

Off the top of my head I have two solutions
1. Leave things where they are but move the implementation into the `cpp` file so that things that depend on `Group.hpp` (everything) don't depend on `pygeosx`.
2. Move all of they python wrapper classes out of `pygeosx` and next to the classes that they wrap. Then we can leave `pygeosx` to contain all the `C++` glue as well as whatever pure python stuff we want to build on top.

Reply:
As discussed during the meeting, I added a subdirectory `python` under the different class we are wrapping with python, and modified/created the `CMakeLists` associated (including the ones for the `unit_tests`). All circular references are now fixed.